File: examples/real-world/buffer_ops.c
// Buffer operations inspired by real systems programming
// Testing: arrays, pointer arithmetic, bounds checking patterns

void buffer_fill(int* buffer, int size, int value) {
    int i;
    for (i = 0; i < size; i = i + 1) {
        buffer[i] = value;
    }
}

int buffer_sum(int* buffer, int size) {
    int sum;
    int i;
    sum = 0;
    for (i = 0; i < size; i = i + 1) {
        sum = sum + buffer[i];
    }
    return sum;
}

int buffer_find(int* buffer, int size, int target) {
    int i;
    for (i = 0; i < size; i = i + 1) {
        if (buffer[i] == target) {
            return i;
        }
    }
    return -1;
}

void buffer_reverse(int* buffer, int size) {
    int i;
    int temp;
    for (i = 0; i < size / 2; i = i + 1) {
        temp = buffer[i];
        buffer[i] = buffer[size - 1 - i];
        buffer[size - 1 - i] = temp;
    }
}

File: examples/real-world/buffer_ops.rs
static mut ERRNO: i32 = 0;
fn buffer_reverse<'a>(mut buffer: &mut [i32]) {
    let mut i: i32 = 0i32;
    let mut temp: i32 = 0i32;
    i = 0;
while i < (buffer.len() as i32 / 2) {
    temp = buffer[(i) as usize];
    buffer[(i) as usize] = buffer[((buffer.len() as i32 - 1) - i) as usize];
    buffer[((buffer.len() as i32 - 1) - i) as usize] = temp;
    i = i + 1;
}
}
fn buffer_find<'a>(mut buffer: &[i32], mut target: i32) -> i32 {
    let mut i: i32 = 0i32;
    i = 0;
while i < buffer.len() as i32 {
    if buffer[(i) as usize] == target {
    return i;
}
    i = i + 1;
}
    return -1;
}
fn buffer_sum<'a>(mut buffer: &[i32]) -> i32 {
    let mut sum: i32 = 0i32;
    let mut i: i32 = 0i32;
    sum = 0;
    i = 0;
while i < buffer.len() as i32 {
    sum = sum + buffer[(i) as usize];
    i = i + 1;
}
    return sum;
}
fn buffer_fill<'a>(mut buffer: &mut [i32], mut value: i32) {
    let mut i: i32 = 0i32;
    i = 0;
while i < buffer.len() as i32 {
    buffer[(i) as usize] = value;
    i = i + 1;
}
}
